cardano-testnet/src/Testnet/Cardano.hs

```diff
@@ -298,98 +378,134 @@ cardanoTestnet testnetOptions H.Conf {..} = do
     , "--delegate-share", "1"
     , "--avvm-entry-count", "0"
     , "--avvm-entry-balance", "0"
-    , "--protocol-parameters-file", tempAbsPath </> "byron.genesis.spec.json"
-    , "--genesis-output-dir", tempAbsPath </> "byron"
+    , "--protocol-parameters-file", tmpDir </> "byron.genesis.spec.json"
+    , "--genesis-output-dir", tmpDir </> "byron"
     ]
 
   H.renameFile
-    (tempAbsPath </> "byron.genesis.spec.json")
-    (tempAbsPath </> "byron/genesis.spec.json")
+    (tmpDir </> "byron.genesis.spec.json")
+    (tmpDir </> "byron/genesis.spec.json")
 
   -- Symlink the BFT operator keys from the genesis delegates, for uniformity
   forM_ bftNodesN $ \n -> do
-    H.createFileLink (tempAbsPath </> "byron/delegate-keys.00" <> show @Int (n - 1) <> ".key") (tempAbsPath </> "node-bft" <> show @Int n </> "byron/delegate.key")
-    H.createFileLink (tempAbsPath </> "byron/delegation-cert.00" <> show @Int (n - 1) <> ".json") (tempAbsPath </> "node-bft" <> show @Int n </> "byron/delegate.cert")
+    H.createFileLink (tmpDir </> "byron/delegate-keys.00" <> show @Int (n - 1) <> ".key") (tmpDir </> "node-bft" <> show @Int n </> "byron/delegate.key")
+    H.createFileLink (tmpDir </> "byron/delegation-cert.00" <> show @Int (n - 1) <> ".json") (tmpDir </> "node-bft" <> show @Int n </> "byron/delegate.cert")
+  let
+    forallBftNodesMkFile x = forM [1..numBftNodes] (fakeItH . x)
+
+  (genesisKeys :: [File ByronKey])<- forallBftNodesMkFile $ \n -> "byron/genesis-keys.00" <> show @Int (n - 1) <> ".json"
```

Comment:
Can the typing be done by type applications instead of type ascriptions?

Reply:
Thanks for spotting, those types can be inferred and annotation are superfluous.